### PR TITLE
feat: add stuck tasks checker

### PR DESCRIPTION
**Pull request by tatsianazh**
Checker allows detecting stuck tasks in ‘running’ (>2 hours) and ‘enqueued’ (>24 hours) status. The maximum run time for both cases is configurable.

### CODE REVIEW

File: model/Check/System/StuckTasksCheckService.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace oat\taoSystemStatus\model\Check\System;
+
+use oat\oatbox\service\ConfigurableService;
+
+class StuckTasksCheckService extends ConfigurableService
+{
+    public const SERVICE_ID = 'taoSystemStatus/TaskQueueCheck';
```
Comment (tikhanovichA) [inline model/Check/System/StuckTasksCheckService.php]: service ID should be equal to the name of the service and config
`taoSystemStatus/StuckTasksCheck`

**Review by tikhanovichA — CHANGES_REQUESTED** [review]

File: model/Check/System/StuckTasksCheckService.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace oat\taoSystemStatus\model\Check\System;
```
Comment (tikhanovichA) [inline model/Check/System/StuckTasksCheckService.php]: let's move that service from the checks folder. For example model/SystemStatus

File: model/Check/System/StuckTasksCheck.php [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    {
+        $message = '';
+        $allStuckTasks = $this->getAllStuckTasks();
+
+        if (empty($allStuckTasks)){
+            $report = new Report(Report::TYPE_SUCCESS, __('There is no stuck tasks'));
+        } else {
+            foreach ($allStuckTasks as $stuckTask) {
+                $task = [
+                    'task_id' => $stuckTask['task_id'],
```
Comment (tikhanovichA) [inline model/Check/System/StuckTasksCheck.php]: task_id and status are reusable strings and can be declared in constants
For example:
const OPTION_TASK_ID = 'task_id';

File: config/default/StuckTasksCheck.conf.php [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+use oat\taoSystemStatus\model\Check\System\StuckTasksCheckService;
+
+return new \oat\taoSystemStatus\model\Check\System\StuckTasksCheckService([
+    StuckTasksCheckService::RUNNING_MAX_TIME => 2, // max time for running status, hours
```
Comment (tikhanovichA) [inline config/default/StuckTasksCheck.conf.php]: let's update it to minutes. Sometimes we would need to check less than 1h

File: model/Check/System/StuckTasksCheck.php [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    private function getEnqueuedTasks()
+    {
+        $enqueuedMaxTime = $this->getStuckTasksCheckService()->getEnqueuedMaxTime();
+        $maxEnqueued = time() - $enqueuedMaxTime * 3600;
+
+        /** @var TaskLogInterface $taskQueueLog */
+        $taskQueueLog = $this->getServiceLocator()->get(TaskLogInterface::SERVICE_ID);
+        $filter = (new TaskLogFilter())
+            ->in(TaskLogBrokerInterface::COLUMN_STATUS, [TaskLogInterface::STATUS_DEQUEUED])
```
Comment (tikhanovichA) [inline model/Check/System/StuckTasksCheck.php]: Maybe STATUS_ENQUEUED ?

File: model/Check/System/StuckTasksCheckService.php (hunk 1 above)
Comment (tatsianazh) [reply model/Check/System/StuckTasksCheckService.php]: fixed

File: model/Check/System/StuckTasksCheckService.php (hunk 2 above)
Comment (tatsianazh) [reply model/Check/System/StuckTasksCheckService.php]: done

File: model/Check/System/StuckTasksCheck.php (hunk 3 above)
Comment (tatsianazh) [reply model/Check/System/StuckTasksCheck.php]: done

File: config/default/StuckTasksCheck.conf.php (hunk 4 above)
Comment (tatsianazh) [reply config/default/StuckTasksCheck.conf.php]: fixed

File: model/Check/System/StuckTasksCheck.php (hunk 5 above)
Comment (tatsianazh) [reply model/Check/System/StuckTasksCheck.php]: fixed

**Review by tikhanovichA — APPROVED** [review]

**Pull request merged**